packages/react-aria-components/src/Table.tsx

```diff
@@ -93,7 +93,7 @@ class TableCollection<T> extends BaseCollection<T> implements ITableCollection<T
   }
 
   get size() {
-    return [...this.getChildren(this.body.key)].length;
+    return [...this.getChildren(this.body.key)].filter(item => item.type !== 'loader').length;
```

Comment:
I think `size` is used to calculate the `aria-rowcount` right? Shouldn't the loader be included in that? It is presented as a row in the table so seems like maybe it should? What else is `size` used for that caused a problem here?

Reply:
Thats right, but I was hoping that we'd modify https://github.com/adobe/react-spectrum/blob/10bac541a2a5c666b443fe70f313455a405a5c77/packages/%40react-aria/table/src/useTable.ts#L71-L73 to essentially do `[...this.getChildren(this.body.key)].length` as part of its calculation. For the most part though, checks on `collection.size` like in https://github.com/adobe/react-spectrum/blob/10bac541a2a5c666b443fe70f313455a405a5c77/packages/%40react-aria/table/src/useTableColumnHeader.ts#L83 and https://github.com/adobe/react-spectrum/blob/10bac541a2a5c666b443fe70f313455a405a5c77/packages/%40react-aria/table/src/useTableSelectionCheckbox.ts#L67 would want to still resolve as if the collection were truly empty even when the loading spinner exists in an empty table.

> What else is size used for that caused a problem here?

The two lines linked above were problematic since the table would still support keyboard navigation when it was in a loading state, which was at odds from previous behavior since the table itself used to be the only tab stop if it was empty. There were some crashes on keyboard nav as well but those are fixed now by the changes in GridKeyboardDelegate. If we are ok with the keyboard nav experience being different between the loading state and the empty state, I can revert this change